modules/consensus/accept.go

```diff
@@ -236,6 +236,11 @@ func (cs *ConsensusSet) managedAcceptBlock(b types.Block) error {
 			// a new block to the cache.
 			if err == errFutureTimestamp {
 				go func() {
+					if cs.tg.Add() != nil {
+						return
+					}
+					defer cs.tg.Done()
+
 					time.Sleep(time.Duration(b.Timestamp-(types.CurrentTimestamp()+types.FutureThreshold)) * time.Second)
```

Comment:
Need to use a `select` here with `time.After`; otherwise the goroutine will run for a long time before shutdown can complete.